### PR TITLE
Rename black-/whitelist to allow/deny-list

### LINKED ISSUE #975: v2: Rename black-/whitelist to allow/deny-list
In all the places in the code and the CLI flags we want to rename the black-/whitelist to allow/deny-list.

Blocked for merging until after the v1.9.0 release.

### PR DESCRIPTION
**Pull request by tlereste**
**What this PR does / why we need it**:
Rename black-/whitelist to allow/deny-list for v2.0.0 release.
**Which issue(s) this PR fixes** :
Fixes #975

### CODE REVIEW

**Comment by tlereste** [comment]
During a bad rebase of https://github.com/kubernetes/kube-state-metrics/pull/1002 my PR was closed, so I recreated this one.

File: pkg/allowdenylist/allowdenylist.go [hunk 1]
```diff
@@ -107,31 +107,31 @@ func (l *WhiteBlackList) IsIncluded(item string) bool {
...
 	}
 
-	if l.isWhiteList {
-		return "whitelisting the following items: " + strings.Join(items, ", ")
+	if l.isAllowList {
+		return "allowlisting the following items: " + strings.Join(items, ", ")
 	}
 
-	return "blacklisting the following items: " + strings.Join(items, ", ")
+	return "denylisting the following items: " + strings.Join(items, ", ")
```
Comment (LiliC) [inline pkg/allowdenylist/allowdenylist.go]: hmm I would reword all the user facing strings to make it more readable, wdyt of something like:
> Excluding the following lists that were on denylist: .....

File: pkg/allowdenylist/allowdenylist.go [hunk 2]
```diff
@@ -107,31 +107,31 @@ func (l *WhiteBlackList) IsIncluded(item string) bool {
...
+func (l *AllowDenyList) Status() string {
 	items := make([]string, 0, len(l.list))
 	for key := range l.list {
 		items = append(items, key)
 	}
 
-	if l.isWhiteList {
-		return "whitelisting the following items: " + strings.Join(items, ", ")
+	if l.isAllowList {
+		return "allowlisting the following items: " + strings.Join(items, ", ")
```
Comment (LiliC) [inline pkg/allowdenylist/allowdenylist.go]: Same as below:
> Including the following lists that were on allowlist: .....

File: pkg/allowdenylist/allowdenylist.go (hunk 1 above)
Comment (tlereste) [reply pkg/allowdenylist/allowdenylist.go]: Done

File: pkg/allowdenylist/allowdenylist.go (hunk 2 above)
Comment (tlereste) [reply pkg/allowdenylist/allowdenylist.go]: Done
